python/ql/test/experimental/dataflow/tainttracking/defaultAdditionalTaintStep/test_syntax.py

```diff
@@ -0,0 +1,45 @@
+# Add taintlib to PATH so it can be imported during runtime without any hassle
+import sys; import os; sys.path.append(os.path.dirname(os.path.dirname((__file__))))
+from taintlib import *
+
+# This has no runtime impact, but allows autocomplete to work
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ..taintlib import *
+
+
+# Actual tests
+
+class Context:
+    def __enter__(self):
+        return TAINTED_STRING
+
+    def __exit__(self, exc_type, exc, tb):
+        pass
+
+def test_with():
+    ctx = Context()
+    taint(ctx)
+    with ctx as tainted:
+        ensure_tainted(tainted) # $ tainted
```

Comment:
I'm not 100% on board with the way this test is designed, due to the fact that you BOTH do `taint(ctx)` and the `__enter__` method returns `TAINTED_STRING` -- which one is `tainted` getting tainted from?

Can we get a test for each of these cases please? :pray:

Reply:
I comes from the `taint` call. The context object being tainted is also what makes one of the new tests work, I think.